observability/src/test/java/io/grpc/observability/interceptors/InternalLoggingChannelInterceptorTest.java

```diff
@@ -156,122 +169,123 @@ public String authority() {
 
     // send client header
     {
-      EventType expectedRequestHeaderEvent = EventType.GRPC_CALL_REQUEST_HEADER;
-      ArgumentCaptor<GrpcLogRecord> captor = ArgumentCaptor.forClass(GrpcLogRecord.class);
       Metadata clientInitial = new Metadata();
       String dataA = "aaaaaaaaa";
       String dataB = "bbbbbbbbb";
       Metadata.Key<String> keyA =
           Metadata.Key.of("a", Metadata.ASCII_STRING_MARSHALLER);
       Metadata.Key<String> keyB =
           Metadata.Key.of("b", Metadata.ASCII_STRING_MARSHALLER);
-      MetadataEntry entryA =
-          MetadataEntry
-              .newBuilder()
-              .setKey(keyA.name())
-              .setValue(ByteString.copyFrom(dataA.getBytes(US_ASCII)))
-              .build();
-      MetadataEntry entryB =
-          MetadataEntry
-              .newBuilder()
-              .setKey(keyB.name())
-              .setValue(ByteString.copyFrom(dataB.getBytes(US_ASCII)))
-              .build();
       clientInitial.put(keyA, dataA);
       clientInitial.put(keyB, dataB);
-      GrpcLogRecord.Metadata expectedMetadata = GrpcLogRecord.Metadata
-          .newBuilder()
-          .addEntry(entryA)
-          .addEntry(entryB)
-          .build();
       interceptedLoggingCall.start(mockListener, clientInitial);
-      verify(mockSink).write(captor.capture());
-      assertEquals(captor.getValue().getEventType(),
-          expectedRequestHeaderEvent);
-      assertEquals(captor.getValue().getSequenceId(), 1L);
-      assertEquals(captor.getValue().getServiceName(), "service");
-      assertEquals(captor.getValue().getMethodName(), "method");
-      assertEquals(captor.getValue().getAuthority(), "the-authority");
-      assertEquals(captor.getValue().getMetadata(), expectedMetadata);
-      verifyNoMoreInteractions(mockSink);
+      verify(mockLogHelper).logRequestHeader(
+          /*seq=*/ eq(1L),
+          eq("service"),
+          eq("method"),
+          eq("the-authority"),
+          ArgumentMatchers.<Duration>isNull(),
+          same(clientInitial),
+          eq(filterParams.headerBytes),
+          eq(EventLogger.LOGGER_CLIENT),
+          anyString(),
+          ArgumentMatchers.<SocketAddress>isNull());
+      verifyNoMoreInteractions(mockLogHelper);
       assertSame(clientInitial, actualClientInitial.get());
     }
 
-    // TODO(dnvindhya) : Add a helper method to verify other fields of GrpcLogRecord for all events
     // receive server header
     {
```

Comment:
Without calling `reset(mockLogHelper)` how are you able to perform more tests in the same test-method?

Reply:
`VerifyNoMoreInteractions()` basically checks if all the methods invoked on the mocked object are verified using `verify`. Since the test verifies all the method invocation test executes successfully. 
Added reset for `mockLogHelper` and `mockListener` after every test block.